integration/presto/src/main/java/org/apache/carbondata/presto/impl/CarbonTableReader.java

```diff
@@ -209,20 +207,21 @@ public TBase create() {
               tablePath);
 
       wrapperTableInfo.setTransactionalTable(isTransactionalTable);
-
-      CarbonMetadata.getInstance().removeTable(wrapperTableInfo.getTableUniqueName());
-      // Step 4: Load metadata info into CarbonMetadata
-      CarbonMetadata.getInstance().loadTableMetadata(wrapperTableInfo);
-      CarbonTable carbonTable = Objects.requireNonNull(
-          CarbonMetadata.getInstance().getCarbonTable(table.getSchemaName(), table.getTableName()),
-          "carbontable is null");
-      // If table is not previously cached, then:
-      if (cache == null) {
-        cache = new CarbonTableCacheModel(modifiedTime, carbonTable);
-        // cache the table
-        carbonCache.get().put(table, cache);
-      } else {
-        cache.setCarbonTable(carbonTable);
+      // multiple tasks can be launched in a worker concurrently. Hence need to synchronize this.
+      synchronized (this) {
```

Comment:
Better keep schema reading also in synchronized block

Reply:
ok. done